python/pyiceberg/catalog/__init__.py

```diff
@@ -600,3 +600,7 @@ def _get_updated_props_and_update_summary(
         )
 
         return properties_update_summary, updated_properties
+
+    def __repr__(self) -> str:
+        """Returns the string representation of the Catalog class."""
+        return f"{self.__class__} with name: {self.name} and properties: {self.properties}"
```

Comment:
I think this is great, but I'm hesitant to print the properties as well because it might expose credentials that might end up in logs.

Reply:
removed that